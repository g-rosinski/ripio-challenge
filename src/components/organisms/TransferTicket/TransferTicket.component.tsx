import React from "react"
import { TransferTransaction } from "../../../models/Transfer"
import { format } from "fecha"
import { TransferTicketContainer } from "./TransferTicket.styles"
import { currencyFormat } from "../../../shared/formatter"
import { LabellyRecord } from "../../molecules/LabellyRecord/LabellyRecord.component"

export type TransferTicketProps = TransferTransaction

export const TransferTicket: React.FC<TransferTicketProps> = ({
    emiter,
    receptor,
    currency,
    amount,
    transfer_date,
    status
}) => (
    <TransferTicketContainer>
        <LabellyRecord 
            label="Se transfiriÃ³" 
            value={`${currency} ${currencyFormat(amount)}`} 
        />
        <LabellyRecord 
            label="Desde la cuenta" 
            value={emiter} 
        />
        <LabellyRecord 
            label="A la cuenta" 
            value={receptor} 
        />
        <LabellyRecord 
            label="Fecha" 
            value={format(transfer_date,'DD/MM/YYYY  HH:mm')} 
        />
        <LabellyRecord 
            label="Estado de la transferencia" 
            value={status} 
            displayDivider={false}
        />
    </TransferTicketContainer>
)
