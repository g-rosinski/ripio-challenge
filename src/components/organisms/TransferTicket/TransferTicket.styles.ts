import styled from "styled-components";

export const TransferTicketContainer = styled.div`
    display: flex;
    flex-direction: column;
    padding: 30px 15px;
    width: 100%;
    min-width: 250px;
    max-width: 600px;
`
