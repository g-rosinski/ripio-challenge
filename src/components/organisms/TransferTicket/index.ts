export * from './TransferTicket.component'
export * from './TransferTicket.styles'