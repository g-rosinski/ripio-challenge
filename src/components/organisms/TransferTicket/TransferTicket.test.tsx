import { render, screen } from "@testing-library/react"
import React from "react"
import { TransferTicket, TransferTicketProps } from "./TransferTicket.component"

describe('<TransferTicket />', () => {

    const baseProps: TransferTicketProps = { 
        emiter: 'emisor@mail.com', 
        receptor: 'receptor@mail.com', 
        currency: 'ETH', 
        amount: 1100,
        transfer_date: new Date(Date.now()),
        status: 'COMPLETED'
    }

    test('Se reenderice Correctamente', () => {
        const { container } = render(<TransferTicket {...baseProps} />)

        expect(screen.getByText('ETH $ 1.100,00')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})