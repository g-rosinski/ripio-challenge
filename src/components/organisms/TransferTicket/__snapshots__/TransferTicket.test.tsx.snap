// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransferTicket /> Se reenderice Correctamente 1`] = `
<div>
  <div
    class="sc-bcXHqe hcRRHK"
  >
    <div
      class="sc-gswNZR cRPCdo"
    >
      <label
        class="sc-eDvSVe MAutU"
      >
        Se transfirió
        :
      </label>
      <label
        class="sc-eDvSVe MAutU"
      >
        ETH $ 1.100,00
      </label>
    </div>
    <hr
      class="sc-jrcTuL ehiTLY"
      role="presentation"
    />
    <div
      class="sc-gswNZR cRPCdo"
    >
      <label
        class="sc-eDvSVe MAutU"
      >
        Desde la cuenta
        :
      </label>
      <label
        class="sc-eDvSVe MAutU"
      >
        emisor@mail.com
      </label>
    </div>
    <hr
      class="sc-jrcTuL ehiTLY"
      role="presentation"
    />
    <div
      class="sc-gswNZR cRPCdo"
    >
      <label
        class="sc-eDvSVe MAutU"
      >
        A la cuenta
        :
      </label>
      <label
        class="sc-eDvSVe MAutU"
      >
        receptor@mail.com
      </label>
    </div>
    <hr
      class="sc-jrcTuL ehiTLY"
      role="presentation"
    />
    <div
      class="sc-gswNZR cRPCdo"
    >
      <label
        class="sc-eDvSVe MAutU"
      >
        Fecha
        :
      </label>
      <label
        class="sc-eDvSVe MAutU"
      >
        28/12/2022  17:25
      </label>
    </div>
    <hr
      class="sc-jrcTuL ehiTLY"
      role="presentation"
    />
    <div
      class="sc-gswNZR cRPCdo"
    >
      <label
        class="sc-eDvSVe MAutU"
      >
        Estado de la transferencia
        :
      </label>
      <label
        class="sc-eDvSVe MAutU"
      >
        COMPLETED
      </label>
    </div>
  </div>
</div>
`;
